lib/BackendClient.js

```diff
@@ -128,11 +126,11 @@ var BackendClient = (function(){
         return deferred.promise;
     }
 
-    function getEntries(list) {
+    function getEntries(list, logger) {
```

Comment:
If we put logger first it will cause a problem whenever someone is trying to call method in the future without passing a logger as a parameter. I think it's an option we should keep open... never know what's gonna be in the future.